ghcide/test/exe/Config.hs

```diff
@@ -52,37 +56,50 @@ dummyPlugin = mkPluginTestDescriptor (\_ pid -> defaultPluginDescriptor pid "dum
 
-runWithDummyPlugin' ::  FS.VirtualFileTree -> (FileSystem -> Session a) -> IO a
-runWithDummyPlugin' = runSessionWithServerInTmpDirCont' def dummyPlugin
+testWithConfig :: String -> TestConfig () -> Session () -> TestTree
+testWithConfig name conf s = testCase name $ runSessionWithTestConfig conf $ const s
 
-runWithDummyPluginAndCap' :: ClientCapabilities -> (FileSystem -> Session ()) -> IO ()
-runWithDummyPluginAndCap' cap = runSessionWithServerAndCapsInTmpDirCont def dummyPlugin cap (mkIdeTestFs [])
-
-testWithDummyPluginAndCap' :: String -> ClientCapabilities -> (FileSystem -> Session ()) -> TestTree
-testWithDummyPluginAndCap' caseName cap = testCase caseName . runWithDummyPluginAndCap' cap
+runWithDummyPlugin' ::  FS.VirtualFileTree -> (FilePath -> Session a) -> IO a
+runWithDummyPlugin' fs = runSessionWithTestConfig def {
+    testPluginDescriptor = dummyPlugin
+    , testDirLocation = Right fs
+    ,  testConfigCaps = lspTestCaps
+    , testShiftRoot = True
+    }
 
 testWithDummyPlugin :: String -> FS.VirtualFileTree -> Session () -> TestTree
 testWithDummyPlugin caseName vfs = testWithDummyPlugin' caseName vfs . const
 
-testWithDummyPlugin' :: String -> FS.VirtualFileTree -> (FileSystem -> Session ()) -> TestTree
+testWithDummyPlugin' :: String -> FS.VirtualFileTree -> (FilePath -> Session ()) -> TestTree
 testWithDummyPlugin' caseName vfs = testCase caseName . runWithDummyPlugin' vfs
 
 testWithDummyPluginEmpty :: String -> Session () -> TestTree
 testWithDummyPluginEmpty caseName = testWithDummyPlugin caseName $ mkIdeTestFs []
 
-testWithDummyPluginEmpty' :: String -> (FileSystem -> Session ()) -> TestTree
+testWithDummyPluginEmpty' :: String -> (FilePath -> Session ()) -> TestTree
 testWithDummyPluginEmpty' caseName = testWithDummyPlugin' caseName $ mkIdeTestFs []
 
-runWithExtraFiles :: String -> (FileSystem -> Session a) -> IO a
+runWithExtraFiles :: String -> (FilePath -> Session a) -> IO a
 runWithExtraFiles dirName action = do
     let vfs = mkIdeTestFs [FS.copyDir dirName]
     runWithDummyPlugin' vfs action
 
-testWithExtraFiles :: String -> String -> (FileSystem -> Session ()) -> TestTree
+testWithExtraFiles :: String -> String -> (FilePath -> Session ()) -> TestTree
 testWithExtraFiles testName dirName action = testCase testName $ runWithExtraFiles dirName action
 
-runInDir :: FileSystem -> Session a -> IO a
-runInDir fs = runSessionWithServerNoRootLock False dummyPlugin def def def (fsRoot fs)
+runInDir :: FilePath -> Session a -> IO a
+runInDir fs = runSessionWithServer def dummyPlugin fs
+
+testSession' :: TestName -> (FilePath -> Session ()) -> TestTree
+testSession' name = testCase name . run'
+
+run :: Session a -> IO a
+run = runSessionWithTestConfig def
+    {testDirLocation=Right (mkIdeTestFs []), testPluginDescriptor=dummyPlugin}
+    . const
+
+run' :: (FilePath -> Session a) -> IO a
+run' = runSessionWithTestConfig def {testDirLocation=Right (mkIdeTestFs []), testPluginDescriptor=dummyPlugin}
```

Comment:
Similar suggestion here

Reply:
I am thinking if we can make our formator to auto format record update